### PR TITLE
Added unit test

### PR DESCRIPTION
**Pull request by pidydx**
[Review on Reviewable](https://reviewable.io/reviews/google/grr/165)

### CODE REVIEW

File: lib/aff4_test.py [hunk 1]
```diff
@@ -435,6 +435,8 @@ def ExerciseAFF4ImageBase(self, classname):
     """Run basic tests on a subclass of AFF4ImageBase."""
     path = "/C.12345/aff4image"
 
+    config_lib.CONFIG.Set("Server.max_unbound_read_size", 1)
```
Comment (destijl) [inline lib/aff4_test.py]: put this down with the test code, so it's more obvious what's happening.

File: lib/aff4_test.py [hunk 2]
```diff
@@ -471,6 +473,9 @@ def ExerciseAFF4ImageBase(self, classname):
     self.assertEqual(len(data), 300)
     self.assertTrue("XXXHello WorldXXX" in data)
     self.assertTrue("XXXYYY" in data)
+    fd.seek(0)
```
Comment (destijl) [inline lib/aff4_test.py]: add a comment about what you're testing and also test the non-failure case that you get 1byte back from a read.

**Comment by pidydx** [comment]
I shuffled the whole thing around and do both tests now.

File: lib/aff4_test.py [hunk 3]
```diff
@@ -471,6 +471,25 @@ def ExerciseAFF4ImageBase(self, classname):
...
     self.assertTrue("XXXYYY" in data)
+    fd.Close()
+
+    # Set the max_unbound_read_size to last size of object at path
+    # before object creation for unbound read() tests.
+    config_lib.CONFIG.Set("Server.max_unbound_read_size", 10100)
+
+    fd = aff4.FACTORY.Create(path, classname, mode="rw", token=self.token)
+    fd.Set(fd.Schema._CHUNKSIZE(10))
+    # Write one byte and verify the unbound read returns 1 byte
```
Comment (grrrrrrrrr) [inline lib/aff4_test.py]: Nit: Comment seems wrong.

**Pull request merged**